sse/src/main/scala/akka/stream/alpakka/sse/EventSource.scala

```diff
@@ -0,0 +1,126 @@
+ * |            |                                           |            |
+ *              |                             Option[String]|
+ * |            |                                           v            |
+ *   +----------o----------+                     +----------o----------+
+ * | | currentLastEventId  |                     |    eventStreams     | |
+ *   +----------o----------+                     +----------o----------+
+ * |            ^                                           |            |
+ *              |           (EventSource, Future[LastEvent])|
+ * |            |                                           v            |
+ *              |                                +----------o----------+
+ * |            +--------------------------------o        unzip        | |
+ *              Future[LastEvent]                +----------o----------+
+ * |                                                        |            |
+ *                                               EventSource|
+ * |                                                        v            |
+ *                                               +----------o----------+
+ * |                                  +----------o       flatten       | |
+ *                     ServerSentEvent|          +---------------------+
+ * |                                  v                                  |
+ *  - - - - - - - - - - - - - - - - - o - - - - - - - - - - - - - - - - -
+ *}}}
+ */
+object EventSource {
+
+  private type LastEvent = Option[ServerSentEvent]
+
+  private val noEvents = Future.successful(Source.empty[ServerSentEvent])
+
+  /**
+   * @param uri URI with absolute path, e.g. "http://myserver/events
+   * @param send function to send a HTTP request
+   * @param lastEventId initial value for Last-Evend-ID header, optional
+   * @param mat implicit `Materializer`
+   * @return source of materialized values of the handler
+   */
+  def apply(
+      uri: Uri,
+      send: HttpRequest => Future[HttpResponse],
+      lastEventId: Option[String] = None
+  )(implicit mat: Materializer): EventStream = {
+
+    val eventStreams = {
+      def getEventStream(lastEventId: Option[String]) = {
+        import EventStreamUnmarshalling._
+        import mat.executionContext
+        val request = {
+          val r = Get(uri).addHeader(Accept(`text/event-stream`))
+          lastEventId.foldLeft(r)((r, i) => r.addHeader(`Last-Event-ID`(i)))
+        }
+        send(request).flatMap(Unmarshal(_).to[EventStream]).fallbackTo(noEvents)
+      }
+      def enrichWithLastElement(eventStream: EventStream) = {
+        val p = Promise[LastEvent]()
+        val enriched =
+          eventStream.alsoToMat(Sink.lastOption)(Keep.both).mapMaterializedValue {
+            case (m, f) =>
+              p.completeWith(f)
+              m
+          }
+        (enriched, p.future)
+      }
+      Flow[Option[String]].mapAsync(1)(getEventStream).map(enrichWithLastElement)
```

Comment:
There can only ever be one request in flight, so I think that `1` is a nice fit. Any other higher number wouldn't change anything.